sweep/tx_input_set.go

```diff
@@ -369,8 +369,14 @@ func (t *txInputSet) addPositiveYieldInputs(sweepableInputs []txInput) {
 }
 
 // tryAddWalletInputsIfNeeded retrieves utxos from the wallet and tries adding
-// as many as required to bring the tx output value above the given minimum.
+// as many as required to bring the tx output value above the given minimum if
+// this is a force sweep set.
 func (t *txInputSet) tryAddWalletInputsIfNeeded() error {
+	// Skip adding wallet input if this is not a force sweep.
+	if !t.force {
```

Comment:
yeah makes sense. but now we need to make sure we force, htlc-timout txns and htlc success txns, otherwise can can run into issue when leading to a forceclose on the incoming channel ?
https://github.com/lightningnetwork/lnd/blob/master/contractcourt/htlc_success_resolver.go#L263-L268 ...

Reply:
We don't set those to `Force` because it makes no sense to sweep them if it's uneconomical, but we still want to add utxo's to sweep them. So I'm not sure if this change makes sense?

Reply:
You are right, incoming HTLCs should only be swept when its economical to do so (that does not apply for outgoing HTLCs where we already Force the sweep because we potentially need to fail back the corresponding incoming htlc). So I agree this approach where we only add utxo when we force a sweep brings us into problems here 🤔